sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -757,7 +758,10 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = {
+    val path = new Path(getConf(WAREHOUSE_PATH))
+    FileSystem.get(path.toUri, new Configuration()).makeQualified(path).toString
```

Comment:
Can we split the difference here and treat those two as different issues?

- for this bug, let's hardcode `.transform(new File(_).toURI().toString())` into the config constant definition; that's on par with the previous version of this code modified in 11a6844beb, which hardcoded the `file:` scheme into the default value.

- separately, if it's a real issue, let's make changes to make this support non-local directories.

That way we fix the immediate issue and can have more discussions about how to properly solve the other and maybe even fix `resolveURI`.